pkg/cliplugins/workspace/plugin/kubeconfig.go

```diff
@@ -305,6 +306,22 @@ func (kc *KubeConfig) currentWorkspace(ctx context.Context, host string, workspa
 	return err
 }
 
+func structureWorkspaceType(workspaceType, currentClusterName string) tenancyv1alpha1.ClusterWorkspaceTypeReference {
+	separatorIndex := strings.LastIndex(workspaceType, ":")
+	switch separatorIndex {
+	case -1:
+		return tenancyv1alpha1.ClusterWorkspaceTypeReference{
+			Name: tenancyv1alpha1.ClusterWorkspaceTypeName(cases.Title(language.Und, cases.NoLower).String(workspaceType)),
+			Path: currentClusterName,
+		}
+	default:
+		return tenancyv1alpha1.ClusterWorkspaceTypeReference{
+			Name: tenancyv1alpha1.ClusterWorkspaceTypeName(cases.Title(language.Und, cases.NoLower).String(workspaceType[separatorIndex+1:])),
+			Path: workspaceType[:separatorIndex],
+		}
+	}
+}
+
```

Comment:
We had other plans with the `--type` flags that overlap with this:

```
$ kubectl ws create --type Foo bar
```

should walk up the hierarchy looking for CWT `foo`. In constrast `--type root:Foo` is an explicit type without the need for this traversal.

I wouldn't mind to automatically derive the right capitalisation in this process.

Reply:
While doing this, how much do we actually stick to the capitalisation. When seeing @davidfestal's Rootbucket, or RootBucket, or RoOTBuckeT, I am not sure our current route is good.